### PR TITLE
Writes templates to directory if --output-dir is set

### LINKED ISSUE #229: "archer app package" experience
### The help menu
```
$ archer app package -h
Prints the CloudFormation template used to deploy an application to an environment.
This command is also run as part of "archer deploy".

Usage
  archer app package [flags]

Flags
  -n, --name string     Name of the application.
  -e --env string       Name of the environment.
  --tag  string         Optional. The application's image tag. Defaults to your latest git commit's hash prefixed with "manual".
  --output-dir string   Optional. Writes the stack template and template configuration to a directory.
  -h, --help            help for package

Examples
  Print the CloudFormation template for the "frontend" application parametrized for the "test" environment.
  $ archer app package -n frontend -e test

  Write the CloudFormation stack and configuration to a "frontend-infra" sub-directory instead of printing.
  $ archer app package -n frontend -e test --output-dir ./frontend-infra
  $ ls ./frontend-infra
  frontend.stack.yml      frontend-test.config.yml
```

### The interactive experience
```
$ archer app package
? Which application's CloudFormation template would you like to generate?
> frontend
> backend
? Which environment's configuration would you like to use for your stack's parameters?
> test
> prod
# Prints the template
```

If the user specified a directory:
```
$ archer app package --output-dir .
? Which application's CloudFormation template would you like to generate?
> frontend
> backend
? Which environment's configuration would you like to use for your stack's parameters?
> test
> prod
✔︎ Success! Wrote the stack template to 'frontend.stack.yml' and template configuration to 'frontend-test.config.yml'
```

## Stories
This command supports two different use-cases:

1. As a user that wants to override a hidden field generated from the manifest,   
I want a command that generates the CFN template and configuration file,   
To find out how the resource names are generated.  
2. Archer's pipeline needs a way of generating the CFN templates and its config files to do deployments.

### PR DESCRIPTION
**Pull request by efekarakus**
Resolves #229 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/cli/app_package.go [hunk 1]
```diff
@@ -157,12 +176,39 @@ func (opts *PackageAppOpts) getTemplate(env *archer.Environment) (string, error)
 			Env:      env,
 			ImageTag: opts.Tag,
 		})
-		return stack.Template()
+		tpl, err := stack.Template()
```
Comment (kohidave) [inline internal/pkg/cli/app_package.go]: What do you think of returning an interface of Template and Params, so you don't need this double string return (dyslexic me has a hard time with double string returns). Or a slice of typed structs {filename, content}

File: internal/pkg/cli/app_package.go [hunk 2]
```diff
@@ -35,33 +39,39 @@ type PackageAppOpts struct {
...
 }
 
 // NewPackageAppOpts returns a new PackageAppOpts where the image tag is set to "manual-{short git sha}".
+// The CloudFormation template is written to stdout and the parameters are discarded by default.
 // If an error occurred while running git, we set the image name to "latest" instead.
 func NewPackageAppOpts() *PackageAppOpts {
 	commitID, err := exec.Command("git", "rev-parse", "--short", "HEAD").CombinedOutput()
-	project := viper.GetString(projectFlag)
 	if err != nil {
 		// If we can't retrieve a commit ID we default the image tag to "latest".
```
[the remark this answers is not in the document]
Comment (efekarakus) [reply internal/pkg/cli/app_package.go]: mm is that bad though? If they deploy multiple times to the same repo with the same commit id then it's fair for us to error I think.
The can always override the value default value by providing a value to the flag.

File: internal/pkg/cli/app_package.go (hunk 1 above)
Comment (efekarakus) [reply internal/pkg/cli/app_package.go]: 👍  I like the latter, created a struct that groups the two templates

File: internal/pkg/cli/app_package.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kohidave) [reply internal/pkg/cli/app_package.go]: Sorry, I meant using latest as the default on line 60 (do I even know how to use GitHub?).

**Pull request merged**